base-theme/assets/webpack/webpack.common.ts

```diff
@@ -122,7 +139,22 @@ const config: webpack.Configuration = {
         openAnalyzer: true
       }) :
       []
-  )
+  ),
+  optimization: {
+    splitChunks: {
+      minChunks:   2,
+      cacheGroups: {
+        common: {
+          test:   /[\\/]node_modules[\\/]/,
+          name:   "common",
+          chunks: chunk => {
+            const splitChunks = [entryNames.www, entryNames.courseV2]
+            return splitChunks.includes(chunk.name)
```

Comment:
This `optimization.splitChunks` bit tells webpack to put anything shared between `www` and `course-v2` entrypoints into a separate chunk, shared between both bundles. The chunk name will be `common.[fullhash].js`. We are responsible for making sure script/link tags are updated appropriately.

Explicitly limited to the `www` and `course-v2` entrypoints to avoid including extra, unecessary stuff. E.g., if we still had `course`(-v1) entrypoint around, then with `minChunks: 2`, VideoJS would have ended up in this chunk, which would be bad for `www`, where it is uncessary.

We could do this more dynamically (more chunks, fewer restrictions), but that complicates the script/link tags. This seems like a good start. See https://github.com/mitodl/ocw-hugo-themes/issues/1033#issuecomment-1401042248 for more details.